app/deepseg.cc

```diff
@@ -199,15 +196,56 @@ class CalcMask final {
 				mask_current = raw_tmp;
 				new_mask = true;
 			}
+			loopns = diffnanosecs(timestamp(), t1);
 		}
 	}
 
+	// timing callbacks
+	static void onprep(void *ctx) {
+		CalcMask *cls = (CalcMask *)ctx;
+		cls->prepns=diffnanosecs(timestamp(), cls->t0);
+		cls->t0 = timestamp();
+	}
+	static void oninfer(void *ctx) {
+		CalcMask *cls = (CalcMask *)ctx;
+		cls->tfltns=diffnanosecs(timestamp(), cls->t0);
+		cls->t0 = timestamp();
+	}
+	static void onmask(void *ctx) {
+		CalcMask *cls = (CalcMask *)ctx;
+		cls->maskns=diffnanosecs(timestamp(), cls->t0);
+		cls->t0 = timestamp();
+	}
+
 public:
-	CalcMask(void *maskctx) :
+	timestamp_t t0;
+	long waitns;
+	long prepns;
+	long tfltns;
+	long maskns;
+	long loopns;
+
+	CalcMask(const char *modelname,
+			 size_t threads,
+			 size_t width,
+			 size_t height) :
 		state{thread_state::INIT},
 		thread{&CalcMask::run, this} {
+		maskctx = bs_maskgen_new(
+					modelname,
+					threads,
+					width,
+					height,
+					nullptr,
+					onprep,
+					oninfer,
+					onmask,
+					this
+					);
```

Comment:
Avoid stretching things onto too many LOC …

Reply:
That was a leftover, i fixed it.